src/game/engine/anim.cpp

```diff
@@ -33,6 +37,34 @@ AnimClass::AnimClass(AnimType type, coord_t coord, unsigned char loop_delay, uns
     m_LoopDelay(loop_delay),
     m_field_4A()
 {
+    if (m_Type->Get_End() == -1) {
+        g_isTheaterShape = m_Type->Is_Theater(); // why is this here? bug?
```

Comment:
It affects which shape buffer the game uses for the subsequent calls to the shape functions.